realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.realm.SyncTestUtils.Companion.createTestUser
+import io.realm.entities.StringOnly
+import junit.framework.Assert.*
+import junit.framework.TestCase
+import org.junit.*
+import org.junit.rules.ErrorCollector
+import org.junit.runner.RunWith
+import java.lang.Exception
+import java.lang.UnsupportedOperationException
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SchemaTests {
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    @get:Rule
+    val errorCollector = ErrorCollector()
+
+    private lateinit var config: SyncConfiguration
+    private lateinit var app: TestRealmApp
+
+    @Before
+    fun setUp() {
+        app = TestRealmApp()
+        val user = createTestUser(app)
+        config = configFactory.createSyncConfigurationBuilder(user).build()
+    }
+
+    @After
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun instance() {
+        val realm = Realm.getInstance(config)
+        realm.use {
+            TestCase.assertFalse(realm.isClosed)
+        }
+        assertTrue(realm.isClosed)
+    }
+
+    @Test
+    fun createObject() {
+        Realm.getInstance(config).use { realm ->
+            realm.beginTransaction()
```

Comment:
Nitpick. You could use `realm.executeTransaction { }` here. I think we started using this in most places since it is the best practise method (since it automatically cancels in case of exceptions).